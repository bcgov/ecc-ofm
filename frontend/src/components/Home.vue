<!-- eslint-disable vue/multi-word-component-names -->
<template>
  <v-card
    title="OFM"
    subtitle="Temp Home Screen"
    text="..."
    style="height:100%;"
  ></v-card>
</template>
