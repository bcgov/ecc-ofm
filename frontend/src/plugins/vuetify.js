import '@fortawesome/fontawesome-free/css/all.css';
import '@mdi/font/css/materialdesignicons.css';
import Vue from 'vue';
import Vuetify from 'vuetify/lib';
import { Ripple } from 'vuetify/lib/directives';

Vue.use(Vuetify, {
  directives: {
    Ripple
  }
});

export default new Vuetify({
  theme: {
    light: true,
    dark: false
  },
  icons: {
    iconfont: 'fa',
    values: {
      login: 'fas fa-user-clock',
      fast: 'fas fa-shipping-fast',
      sign_in: 'fas fa-sign-in-alt',
      info1: 'fas fa-info-circle',
      downArrow: 'fas fa-angle-down',
      upArrow: 'fas fa-angle-up',
      user: 'far fa-user',
      copy: 'fas fa-copy',
      search: 'fas fa-search',
      error: 'fas fa-exclamation-triangle',
      lock: 'fas fa-lock',
      info2: 'fas fa-info-circle fa-10x',
      question: 'fas fa-question-circle fa-10x',
      plus: 'fas fa-plus-circle',
      minus: 'fas fa-minus-circle',
    }
  }
});
