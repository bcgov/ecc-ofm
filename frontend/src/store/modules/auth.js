import ApiService from '@/common/apiService';
/* TODO: uncomment during integration with authservice...
import AuthService from '@/common/authService';
*/
import { Routes } from '@/utils/constants';
import {defineStore} from 'pinia';

export const authStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    acronyms: [],
    isAuthenticated: true  /* TODO: during integration with backend, replace true with... localStorage.getItem('jwtToken') !== null */,
    isAuthorizedUser: true /* TODO: during integration with backend, replace true with... localStorage.getItem('isAuthorizedUser') !== null */,
    userInfo: false,
    isValidChildCareProviderUser: localStorage.getItem('iisValidChildCareProviderUser') !== null,
    isValdProgramUser: localStorage.getItem('isValdProgramUser') !== null,
    isValidFinancialOpsUser: localStorage.getItem('isValidFinancialOpsUser') !== null,    
  }),
  getters: {
    acronymsGet: state => state.acronyms,
    isAuthenticatedGet: state => state.isAuthenticated,
    jwtTokenGet: () => localStorage.getItem('jwtToken'),
    userInfoGet: state => state.userInfo,
    //TODO: 3 temp roles ('CCP_ROLE', 'OPS_ROLE', 'PCM_ROLE') were created in auth.js (loosely
    //based on OFM requirements) for the purpose of achieving a 1st draft of the frontend that
    //will render a home screen and menu with minimal errors given no authorization/backend integration.
    //Thus the following related code is only temporarly and expected to be replaced...
    CCP_ROLE: state => state.isValidChildCareProviderUser,
    OPS_ROLE: state => state.isValdProgramUser,
    PCM_ROLE: state => state.isValidFinancialOpsUser
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true;
        localStorage.setItem('jwtToken', token);
      } else {
        this.isAuthenticated = false;
        localStorage.removeItem('jwtToken');
      }
    },
    async setAuthorizedUser(isAdminUser){
      if (isAdminUser) {
        this.isAuthorizedUser = true;
        localStorage.setItem('isAuthorizedUser', 'true');
      } else {
        this.isAuthorizedUser = false;
        localStorage.removeItem(('isAuthorizedUser'));
      }
    },
    async setAuthorizedWebsocketUser(isAuthorizedWebsocketUser){
      if (isAuthorizedWebsocketUser) {
        this.isAuthorizedWebsocketUser = true;
        localStorage.setItem('isAuthorizedWebsocketUser', 'true');
      } else {
        this.isAuthorizedWebsocketUser = false;
        localStorage.removeItem(('isAuthorizedWebsocketUser'));
      }
    },
    async setUserInfo(userInf) {
      if (userInf) {
        this.userInfo = userInf;
      } else {
        this.userInfo = null;
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setChildCareProviderUser(isValidChildCareProviderUser) {
      if (isValidChildCareProviderUser) {
        this.isValidChildCareProviderUser = true;
        localStorage.setItem('isValidChildCareProviderUser', 'true');
      } else {
        this.isValidChildCareProviderUser = false;
        localStorage.removeItem(('isValidChildCareProviderUser'));
      }
    },
    // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setProgramUser(isValdProgramUser) {
      if (isValdProgramUser) {
        this.isValdProgramUser = true;
        localStorage.setItem('isValdProgramUser', 'true');
      } else {
        this.isValdProgramUser = false;
        localStorage.removeItem(('isValdProgramUser'));
      }
    },
      // TODO: method below is part of 1st draft of app framework and likely to be removed...
    async setFinancialOpsUser(isValidFinancialOpsUser) {
      if (isValidFinancialOpsUser) {
        this.isValidFinancialOpsUser = true;
        localStorage.setItem('isValidFinancialOpsUser', 'true');
      } else {
        this.isValidFinancialOpsUser = false;
        localStorage.removeItem(('isValidFinancialOpsUser'));
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logoutState() {
      localStorage.removeItem('jwtToken');
      this.userInfo = false;
      this.isAuthenticated = false;
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken');

      if(!token){
        await this.getJwtToken();
      }

      if(localStorage.getItem('jwtToken')) {
        const response = await ApiService.apiAxios.get(Routes.USER);
        await this.setUserInfo(response.data);
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken');
        if (this.isAuthenticated && !!token) {
          /* TODO: uncomment during integration with authservice/backend...
          const response = await AuthService.refreshAuthToken(token);
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        } else {
          /* TODO: uncomment during integration with backend...
          const response = await AuthService.getAuthToken();
          */
         //TODO: uncomment 'response' below during integration with backend...
          await this.setAuthorizations(/*response*/);
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null);
        throw e;
      }
    },
    //TODO: uncomment 'response' below during integration with authorization/backend...
    async setAuthorizations(/*response*/) {
    /* TODO: uncomment during integration with backend...
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend);
      }

      ApiService.setAuthHeader(response.jwtFrontend);
    */
      await this.setAuthorizedUser(true);
      await this.setChildCareProviderUser(true);
      await this.setProgramUser(true);
      await this.setFinancialOpsUser(true);
    }
  }
});
