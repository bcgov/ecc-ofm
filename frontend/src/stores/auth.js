import moment from 'moment'
import { defineStore } from 'pinia'

import ApiService from '@/common/apiService'
import AuthService from '@/common/authService'
import { useAppStore } from '@/stores/app'
import { APPLICATION_INTAKE_TYPES, OFM_PROGRAM_CODES, ROLES } from '@/utils/constants'

export const useAuthStore = defineStore('auth', {
  namespaced: true,
  state: () => ({
    isAuthenticated: localStorage.getItem('jwtToken') !== null,
    isMinistryUser: false,
    isImpersonating: false,
    isUserInfoLoaded: false,
    userInfo: null,
    impersonateId: null,
    currentFacility: {},
    permissions: [],
  }),
  getters: {
    isActingProvider: (state) => !state.isMinistryUser || state.isImpersonating,
    hasFacilities: (state) => state.userInfo?.facilities?.length > 0,
    hasPermission: (state) => {
      return (permission) => {
        return state.permissions?.some((p) => p === permission)
      }
    },
  },
  actions: {
    //sets Json web token and determines whether user is authenticated
    async setJwtToken(token = null) {
      if (token) {
        this.isAuthenticated = true
        localStorage.setItem('jwtToken', token)
      } else {
        this.isAuthenticated = false
        localStorage.removeItem('jwtToken')
      }
    },
    //sets the token required for refreshing expired json web tokens
    async logout() {
      localStorage.removeItem('jwtToken')
      this.userInfo = null
      this.isAuthenticated = false
    },
    async getUserInfo() {
      const token = localStorage.getItem('jwtToken')
      if (!token) {
        await this.getJwtToken()
      }
      if (localStorage.getItem('jwtToken')) {
        if (!this.isUserInfoLoaded) {
          const appStore = useAppStore()
          let userInfoRes = undefined
          if (this.impersonateId && this.isMinistryUser) {
            userInfoRes = await ApiService.getUserImpersonateInfo(this.impersonateId)
            this.isImpersonating = true
          } else {
            userInfoRes = await ApiService.getUserInfo()
            this.isMinistryUser = userInfoRes.data.isMinistryUser
            delete userInfoRes.data.isMinistryUser
          }
          this.userInfo = userInfoRes.data
          // Default the facility
          if (this.isActingProvider && this.hasFacilities) {
            this.currentFacility = this.userInfo.facilities[0]
          }
          
          // Lookup the permissions
          let role
          if (this.isImpersonating) {
            // When impersonating always use 'Impersonate', not the impersonated user's role
            role = appStore.roles.find((role) => role.roleName === ROLES.IMPERSONATE)
          } else {
            role = appStore.roles.find((role) => role.roleId === this.userInfo.role?.ofm_portal_roleid)
          }
          this.permissions = role?.permissions.map((p) => p.permissionName)

          /*
            A facility can apply for a core application if it satisfies these conditions:
            1. There is an Open Intake or the facility is in the allowed facility list of a Limited Intake
            2. If the facility is in the CCOF program, their program start date must >= 1 year from the current date.
          */
          this.userInfo?.facilities?.forEach((facility) => {
            facility.intakeWindowCheckForAddApplication = appStore.applicationIntakes?.some((intake) => {
              const isWithinApplicationIntakeWindow = moment().isSameOrAfter(moment(intake.startDate)) && moment().isSameOrBefore(moment(intake.endDate))
              const isOpenIntake = intake.type === APPLICATION_INTAKE_TYPES.OPEN_INTAKE
              const limitedIntakeFacilities = intake?.facilities?.map((facility) => facility.facilityId)
              return isWithinApplicationIntakeWindow && (isOpenIntake || limitedIntakeFacilities?.includes(facility.facilityId))
            })
            facility.ccofEnrolmentCheckForAddApplication = facility?.programCode !== OFM_PROGRAM_CODES.CCOF || (facility?.programCode === OFM_PROGRAM_CODES.CCOF && facility?.ccofOneYearEnrolment)
          })

          this.isUserInfoLoaded = true
        }
      }
    },
    //retrieves the json web token from local storage. If not in local storage, retrieves it from API
    async getJwtToken() {
      try {
        const token = localStorage.getItem('jwtToken')
        if (this.isAuthenticated && !!token) {
          const response = await AuthService.refreshAuthToken(token)
          await this.setAuthorizations(response)
        } else {
          const response = await AuthService.getAuthToken()
          await this.setAuthorizations(response)
        }
      } catch (e) {
        // Remove tokens from localStorage and update state
        await this.setJwtToken(null)
        throw e
      }
    },
    async setAuthorizations(response) {
      if (response.jwtFrontend) {
        await this.setJwtToken(response.jwtFrontend)
      }
      ApiService.setAuthHeader(response.jwtFrontend)
    },
  },
})
