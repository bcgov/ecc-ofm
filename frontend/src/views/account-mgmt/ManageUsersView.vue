<template>
  <v-container fluid>
    <v-row>
      <v-col class="pt-0">
        <h3>Manage users</h3>
      </v-col>
    </v-row>
    <v-row>
      <v-col cols="auto">
        <AppButton @click="toggleShowFilter()" variant="text" :disabled="isLoading">
          Filter by facility
          <v-icon right>mdi-filter</v-icon>
        </AppButton>
      </v-col>
      <v-col cols="3">
        <v-text-field v-if="showFilterInput" v-model="facilityNameFilter" placeholder="Filter by facility name" variant="outlined" density="compact" :disabled="isLoading"></v-text-field>
      </v-col>
      <v-col class="d-flex justify-end align-end">
        <AppButton variant="text" @click="toggleDialog({})" :disabled="isLoading">
          <v-icon left>mdi-plus</v-icon>
          Add new user
        </AppButton>
      </v-col>
    </v-row>
    <v-row>
      <v-col>
        <!-- Users Table -->
        <v-skeleton-loader :loading="isLoading" type="table-tbody">
          <v-data-table :headers="headersUsers" :items="filteredUserFacilities" item-key="contactId" item-value="contactId" show-expand density="compact" :expanded.sync="expanded">
            <!-- Slot to customize expand row event -->
            <template v-slot:item.data-table-expand="{ item }">
              <AppButton @click.stop="toggleExpand(item)" variant="text">
                {{ expanded[0] == item.contactId ? 'hide detail' : 'view' }}
              </AppButton>
            </template>
            <template v-slot:item.actions="{ item }">
              <AppButton @click.stop="toggleDialog(item)" variant="text">edit</AppButton>
            </template>
            <!-- Slots to translate specific column values into display values -->
            <template v-slot:item.role="{ item }">
              <span>{{ getRoleNameById(item.role) }}</span>
            </template>
            <template v-slot:item.isExpenseAuthority="{ item }">
              <span>{{ isExpenseAuthority(item) }}</span>
            </template>
            <template v-slot:item.stateCode="{ item }">
              <span>{{ getStatusDescription(item) }}</span>
            </template>

            <!-- Slot to customize expand row content -->
            <template v-slot:expanded-row="{ columns, item }">
              <tr>
                <td></td>
                <td colspan="6" class="pl-0">
                  <v-row>
                    <v-col cols="11" class="pt-5 pb-0">
                      <h4>Current facility access</h4>
                    </v-col>
                  </v-row>
                  <v-row>
                    <v-col cols="12" class="pt-0 pb-0">
                      <!-- Facilities table -->
                      <v-data-table :headers="headersFacilities" :items="item.facilities" item-key="facilityId" density="compact">
                        <template v-slot:item.address="{ item }">{{ item.address }}, {{ item.city }}</template>
                        <template v-slot:item.isExpenseAuthority="{ item }">{{ item.isExpenseAuthority ? 'Yes' : 'No' }}</template>
                        <template v-slot:bottom><!-- no paging --></template>
                      </v-data-table>
                    </v-col>
                    <v-col cols="12"></v-col>
                  </v-row>
                </td>
                <td class="pl-0">
                  <AppButton v-if="showDeactivateUserButton(item)" variant="text" @click="toggleDeactivateUserDialog(item)">Deactivate user</AppButton>
                </td>
              </tr>
            </template>
          </v-data-table>
        </v-skeleton-loader>
      </v-col>
    </v-row>
    <ManageUserDialog :show="showManageUserDialog" :updatingUser="userToUpdate" @close="toggleDialog" @close-refresh="closeDialogAndRefresh" @update-success-event="updateSuccessEvent" />
    <DeactivateUserDialog :show="showDeactivateUserDialog" :user="userToDeactivate" @close="toggleDeactivateUserDialog" @deactivate="getUsersAndFacilities" />
    <AppButton class="mt-2" id="back-home-button" :primary="false" size="large" width="200px" :to="{ name: 'home' }">&larr; Back to Home</AppButton>
  </v-container>
</template>

<script>
import { mapState } from 'pinia'
import { useAppStore } from '@/stores/app'
import AppButton from '@/components/ui/AppButton.vue'
import { useAuthStore } from '@/stores/auth'
import rolesMixin from '@/mixins/rolesMixin.js'
import alertMixin from '@/mixins/alertMixin'
import { CRM_STATE_CODES } from '@/utils/constants'
import { ApiRoutes } from '@/utils/constants'
import ApiService from '@/common/apiService'
import ManageUserDialog from '@/components/account-mgmt/ManageUserDialog.vue'
import DeactivateUserDialog from '@/components/account-mgmt/DeactivateUserDialog.vue'

export default {
  components: { AppButton, ManageUserDialog, DeactivateUserDialog },
  mixins: [rolesMixin, alertMixin],
  data() {
    return {
      isLoading: false,
      showFilterInput: false,
      facilityNameFilter: '',
      usersAndFacilities: null,
      expanded: [],
      showManageUserDialog: false,
      showDeactivateUserDialog: false,
      editedIndex: -1,
      userToUpdate: {},
      userToDeactivate: {},
      headersUsers: [
        { title: '', key: 'data-table-expand', width: '87px' },
        { title: '', key: 'actions', width: '30px' },
        { title: 'First Name', key: 'firstName', width: '10%' },
        { title: 'Last Name', key: 'lastName', width: '10%' },
        { title: 'Email', key: 'email', width: '12%' },
        { title: 'BCeID', key: 'userName', width: '12%' },
        { title: 'Role', key: 'role', width: '12%' },
        { title: 'Expense Authority', key: 'isExpenseAuthority', width: '12%' },
        { title: 'Status', key: 'stateCode', width: '16%' },
      ],
      headersFacilities: [
        { title: 'Facility Name', key: 'facilityName', width: '35%' },
        { title: 'Address', key: 'address', width: '45%' },
        { title: 'Expense Authority', key: 'isExpenseAuthority', width: '15%' },
      ],
    }
  },
  computed: {
    ...mapState(useAppStore, ['getRoleNameById']),
    ...mapState(useAuthStore, ['userInfo']),

    filteredUserFacilities() {
      this.isLoading = true
      try {
        if (!this.facilityNameFilter) return this.sortUsers(this.usersAndFacilities)
        return this.sortUsers(this.usersAndFacilities.filter((user) => user.facilities.some((facility) => facility.facilityName.toLowerCase().includes(this.facilityNameFilter.toLocaleLowerCase()))))
      } catch (error) {
        this.setFailureAlert('Failed to filter users by facility name', error)
      } finally {
        this.isLoading = false
      }
    },
  },
  async created() {
    await this.getUsersAndFacilities()
  },
  methods: {
    isDeactivatedUser(user) {
      return user?.facilities?.length === 0 && !user?.role
    },

    isSameUser(user) {
      return user?.userName === this.userInfo?.userName
    },

    showDeactivateUserButton(user) {
      return !this.isDeactivatedUser(user) && !this.isSameUser(user) && user?.stateCode === CRM_STATE_CODES.ACTIVE
    },

    /**
     * Get the list of users and facilities
     */
    async getUsersAndFacilities() {
      try {
        this.isLoading = true
        const res = await ApiService.apiAxios.get(ApiRoutes.USER_PERMISSIONS_FACILITIES + '/' + this.userInfo.organizationId)
        this.usersAndFacilities = res.data
      } catch (error) {
        this.setFailureAlert('Failed to get the list of users by organization id: ' + this.userInfo.organizationId, error)
      } finally {
        this.isLoading = false
      }
    },

    /**
     * Toggle the facility input filter
     */
    toggleShowFilter() {
      if (this.showFilterInput) {
        this.showFilterInput = false
        this.facilityNameFilter = ''
      } else {
        this.showFilterInput = true
      }
    },

    /**
     * Toggle expanded row
     */
    toggleExpand(item) {
      const index = this.expanded.indexOf(item.contactId)
      if (index > -1) {
        // Collapse if already expanded
        this.expanded.splice(index, 1)
      } else {
        // Expand the clicked row
        this.expanded = [item.contactId]
      }
    },

    /**
     * Get the status description for a status code
     */
    getStatusDescription(user) {
      if (this.isDeactivatedUser(user) || user?.stateCode === CRM_STATE_CODES.INACTIVE) {
        return 'Inactive'
      }
      return 'Active'
    },

    /**
     * Sort users by: account management role 1st, expense authority 2nd, last name 3rd
     */
    sortUsers(users) {
      return users.sort((a, b) => {
        // Check for account management role and sort by it, with true values first
        const roleA = a.role === this.ROLES.ACCOUNT_MANAGEMENT
        const roleB = b.role === this.ROLES.ACCOUNT_MANAGEMENT
        if (roleA && !roleB) return -1
        if (!roleA && roleB) return 1

        // If roles are the same, then sort by stateCode
        if (a.stateCode !== b.stateCode) {
          return a.stateCode - b.stateCode
        }

        // If stateCode is the same, then sort by lastName
        return a.lastName.localeCompare(b.lastName)
      })
    },

    /**
     * Toggles the manage user dialog opened/closed while passing in the user to edit.
     */
    toggleDialog(user) {
      this.editedIndex = this.usersAndFacilities.indexOf(user)
      this.userToUpdate = Object.assign({}, user)
      this.showManageUserDialog = !this.showManageUserDialog
    },

    toggleDeactivateUserDialog(user) {
      this.showDeactivateUserDialog = !this.showDeactivateUserDialog
      this.userToDeactivate = user
    },

    /**
     * Close the manage user dialog and refresh the user list.
     */
    closeDialogAndRefresh() {
      this.showManageUserDialog = false
      this.getUsersAndFacilities()
    },

    /**
     * Called when the manage user dialog emits a user update was success or fail.
     */
    async updateSuccessEvent(isSuccess, error) {
      if (isSuccess) {
        this.setSuccessAlert('User updated successfully.')
        // A user has been updated, get the latest data for UI.
        await this.getUsersAndFacilities()
      } else {
        this.setFailureAlert('User update failed.', error)
      }
    },

    /**
     * Checks if the user is an Expense Authority for any facility.
     */
    isExpenseAuthority(user) {
      return user.facilities.some((facility) => facility.isExpenseAuthority) ? 'Yes' : 'No'
    },
  },
}
</script>

<style scoped>
:deep(div.v-data-table-footer) {
  padding-top: 20px !important;
}
</style>
