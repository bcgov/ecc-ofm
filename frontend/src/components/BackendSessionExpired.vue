<template>
  <div style="display: none">
    <a id="logout_href" :href="link" />
  </div>
</template>

<script>
import { Routes } from '@/utils/constants'
import { useAuthStore } from '@/stores/auth'
import { mapActions } from 'pinia'
export default {
  name: 'BackendSessionExpired',
  data() {
    return {
      link: Routes.SESSION_EXPIRED
    }
  },
  mounted() {
    window.location = document.getElementById('logout_href').href
  },
  methods: {
    ...mapActions(useAuthStore, ['setJwtToken']),
    logout() {
      this.setJwtToken()
    }
  }
}
</script>
