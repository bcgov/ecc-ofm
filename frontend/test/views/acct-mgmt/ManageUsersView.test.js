import { describe, expect, test } from 'vitest'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'

import { useAuthStore } from '@/stores/auth.js'
import ManageUsersView from '@/views/account-mgmt/ManageUsersView.vue'
import { createTestingPinia } from '@pinia/testing'
import { mount } from '@vue/test-utils'

const vuetify = createVuetify({ components, directives })

describe('Manage Users screen', () => {
  const global = { global: { plugins: [createTestingPinia(), vuetify] } }
  const authStore = useAuthStore()

  test('BceID User', async () => {
    authStore.userInfo = { organziationId: '6fbddbd7-5f8e-ee11-8179-000d3a09d699' }
    const wrapper = mount(ManageUsersView, global)
    console.log(wrapper.html())
    expect(wrapper.html()).toMatchSnapshot()
  })

  /*  it('shows "Account Management" in the first row of the Role column', async () => {
    await flushPromises() // Ensure all rendering is completed
    const firstRow = wrapper.find('table tbody tr:first-child')
    const roleCell = firstRow.find('td:nth-child(5)')

    expect(roleCell.text()).toContain('Account Management')
  }) */
})
