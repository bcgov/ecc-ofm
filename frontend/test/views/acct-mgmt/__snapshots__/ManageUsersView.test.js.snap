// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Manage Users screen > BceID User 1`] = `
"<div data-v-fca923b3=\\"\\" class=\\"v-container v-container--fluid v-locale--is-ltr\\">
  <div data-v-fca923b3=\\"\\" class=\\"v-row\\">
    <div data-v-fca923b3=\\"\\" class=\\"v-col pt-0\\">
      <h3 data-v-fca923b3=\\"\\">Manage users</h3>
    </div>
  </div>
  <div data-v-fca923b3=\\"\\" class=\\"v-row\\">
    <div data-v-fca923b3=\\"\\" class=\\"v-col v-col-auto\\"><button data-v-d22d1084=\\"\\" data-v-fca923b3=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--disabled v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text px-2 text-app-button\\" disabled=\\"\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
        <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"> Filter by facility <i data-v-fca923b3=\\"\\" class=\\"mdi-filter mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" right=\\"\\"></i></span>
        <!--v-if--></span>
        <!---->
        <!---->
      </button></div>
    <div data-v-fca923b3=\\"\\" class=\\"v-col v-col-3\\">
      <!--v-if-->
    </div>
    <div data-v-fca923b3=\\"\\" class=\\"v-col d-flex justify-end align-end\\"><button data-v-d22d1084=\\"\\" data-v-fca923b3=\\"\\" type=\\"button\\" class=\\"v-btn v-btn--disabled v-theme--light v-btn--density-default v-btn--size-default v-btn--variant-text px-2 text-app-button\\" disabled=\\"\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
        <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i data-v-fca923b3=\\"\\" class=\\"mdi-plus mdi v-icon notranslate v-theme--light v-icon--size-default\\" aria-hidden=\\"true\\" left=\\"\\"></i> Add new user </span>
        <!--v-if--></span>
        <!---->
        <!---->
      </button></div>
  </div>
  <div data-v-fca923b3=\\"\\" class=\\"v-row\\">
    <div data-v-fca923b3=\\"\\" class=\\"v-col\\">
      <!-- Users Table -->
      <div data-v-fca923b3=\\"\\" class=\\"v-skeleton-loader v-theme--light\\" aria-busy=\\"true\\" aria-live=\\"polite\\" aria-label=\\"Loading...\\" role=\\"alert\\">
        <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-tbody\\">
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
          <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row-divider\\">
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__table-row\\">
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
              <div class=\\"v-skeleton-loader__bone v-skeleton-loader__text\\"></div>
            </div>
            <div class=\\"v-skeleton-loader__bone v-skeleton-loader__divider\\"></div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div data-v-fca923b3=\\"\\" class=\\"v-container v-locale--is-ltr\\">
    <!---->
    <!---->
    <!---->
    <!---->
  </div>
  <!---->
  <!----><a data-v-d22d1084=\\"\\" data-v-fca923b3=\\"\\" class=\\"v-btn v-btn--elevated v-theme--light v-btn--density-default v-btn--size-large v-btn--variant-elevated px-2 BC-Gov-SecondaryButton text-app-button mt-2 mt-2\\" style=\\"width: 400px;\\"><span class=\\"v-btn__overlay\\"></span><span class=\\"v-btn__underlay\\"></span>
    <!----><span class=\\"v-btn__content\\" data-no-activator=\\"\\"><span data-v-d22d1084=\\"\\" class=\\"button-text overflow-hidden\\"><i class=\\"mdi-arrow-left mdi v-icon notranslate v-theme--light v-icon--size-default pb-1\\" aria-hidden=\\"true\\"></i> Back to Account Management</span>
    <!--v-if--></span>
    <!---->
    <!---->
  </a>
</div>"
`;
